src/packet/key/secret.rs

```diff
@@ -39,6 +39,75 @@ impl<D> Drop for SecretKeyInner<D> {
     }
 }
 
+impl<D: PublicKeyTrait + crate::ser::Serialize> SecretKeyInner<D> {
+    /// Remove the password protection of the private key material in this secret key packet.
+    /// This permanently "unlocks" the secret key material.
+    ///
+    /// If the Secret Key material in the packet is not locked, it is left unchanged.
+    ///
+    /// The current locking password for this key must be provided in `password`.
```

Comment:
the docs should be on the public version, as these will not show up for users

Reply:
good point.

moved the password-related functions into a trait, in part to have a shared place for the docs.
i'm very open for thoughts and bikeshedding :)

Reply:
why make this a new trait, instead of adding to the existing `SecretKeyTrait`?

Reply:
These password operations are kind of specific to Secret Key Packets.
They don't apply to some other implementations of `SecretKeyTrait`, such as HSM-backed keys

Reply:
Maybe the trait should be named `SecretKeyPacketProtection`

Reply:
then lets not use a trait but impl them directly, I don’t think it’s good to have an api with a bunch of traits

Reply:
That way we'll have to duplicate the doc comments, but I'm not super opposed to the approach